libGDX/core/src/com/mygdx/game/movement/Board.java

```diff
@@ -59,10 +59,33 @@ public void movePiece(Piece piece, int oldX, int oldY, int newX, int newY) {
     cells[newX][newY].placePiece(p);
   }
 
+  public void movePiece(CoOrdinatePair from, CoOrdinatePair to) {
+    getCell(from).ifPresentOrElse(fromCell -> {
+      fromCell.getPiece().ifPresentOrElse(piece -> {
+        fromCell.removePiece();
+        getCell(to).ifPresentOrElse(cell -> cell.placePiece(piece), () -> {
+          throw new RuntimeException(to + ", cell does not exist on board");
```

Comment:
`cell.placePiece(piece)` will throw an exception if there is a Piece on that cell already. Need to remove the Piece on the cell first.

Reply:
This was done in the Player class, but I have addressed this, and have added it here.